fix: v0.21.0 릴리즈 CI 실패 — PreloadContent 테스트 root 권한 문제

## 개요
v0.21.0 릴리즈 CI가 실패했습니다.

## CI 로그
```
--- FAIL: TestPreloadContent/preload_read_error_falls_back_to_nil_content (0.00s)
    scanner_test.go:1298: expected Content to be nil for unreadable file
FAIL	github.com/indigo-net/Brf.it/pkg/scanner	0.022s
```

## 원인
CI 환경에서 root 권한으로 실행되어 `chmod 0000` 후에도 `os.ReadFile`이 성공합니다. 
테스트가 non-root 환경(로컬)에서만 통과하고 CI(root)에서 실패합니다.

## 환경
- Workflow: release.yml
- Run ID: 23125646110
- Tag: v0.21.0

## 다음 단계
1. 테스트를 root 환경에서도 안전하게 동작하도록 수정 (t.Skip if root)
2. 수정 커밋 후 새 태그(v0.21.0 삭제 → 재태그 또는 v0.21.1)로 릴리즈 재시도

--- a/pkg/scanner/scanner_test.go
+++ b/pkg/scanner/scanner_test.go
@@ -1261,6 +1261,9 @@ func TestPreloadContent(t *testing.T) {
 	})
 
 	t.Run("preload read error falls back to nil content", func(t *testing.T) {
+		if os.Getuid() == 0 {
+			t.Skip("skipping: chmod 0000 has no effect when running as root")
+		}
 		// Create a file then make it unreadable to trigger ReadFile failure.
 		unreadable := filepath.Join(tmpDir, "unreadable.go")
 		if err := os.WriteFile(unreadable, []byte("package main\n"), 0644); err != nil {

